src/CoreWCF.Http/src/CoreWCF/Channels/HttpRequestContext.cs

```diff
@@ -375,10 +382,34 @@ public override HttpOutput GetHttpOutput(Message message)
                 return HttpOutput.CreateHttpOutput(_aspNetContext, HttpTransportSettings, message, HttpMethod);
             }
 
-            protected override SecurityMessageProperty OnProcessAuthentication()
+            protected override async Task<SecurityMessageProperty> OnProcessAuthenticationAsync()
             {
-                // TODO: Wire up authentication for ASP.Net Core
-                //return Listener.ProcessAuthentication(listenerContext);
+                if (HttpTransportSettings.IsAuthenticationRequired)
+                {
+                    AuthenticateResult authenticateResult = _aspNetContext.Features.Get<AuthenticateResult>();
+                    if (authenticateResult == null)
+                    {
+                        // No authentication occurred
+                        return null;
+                    }
+
+                    if (authenticateResult.None)
+                    {
```

Comment:
This code is unreachable as RequestDelegateHandler.HandleRequest checks this and calls ChallengeAsync if it's true.